core/src/main/scala/org/http4s/headers/Content-Type.scala

```diff
@@ -30,14 +30,12 @@ object `Content-Type` {
     ParseResult.fromParser(parser, "Invalid Content-Type header")(s)
 
   private[http4s] val parser: Parser[`Content-Type`] =
-    (MediaRange.parser ~ MediaRange.mediaTypeExtensionParser.rep0).map {
+    (MediaRange.parser ~ MediaRange.mediaTypeExtensionParser.rep0).flatMap {
```

Comment:
`mapFilter` is more efficient than `flatMap` in cats-parse, but wouldn't give as nice an error message.  I think we could do it with a variant of `select` and get that `failWith` message.